diff --git a/node_modules/@bam.tech/react-native-image-resizer/package.json b/node_modules/@bam.tech/react-native-image-resizer/package.json
index 7a73b6e..2033e59 100644
--- a/node_modules/@bam.tech/react-native-image-resizer/package.json
+++ b/node_modules/@bam.tech/react-native-image-resizer/package.json
@@ -169,12 +169,8 @@
     ]
   },
   "codegenConfig": {
-    "libraries": [
-      {
-        "name": "RNImageResizerSpec",
-        "type": "modules",
-        "jsSrcsDir": "src"
-      }
-    ]
+    "name": "RNImageResizerSpec",
+    "type": "modules",
+    "jsSrcsDir": "src"
   }
 }
diff --git a/node_modules/@bam.tech/react-native-image-resizer/react-native-image-resizer.podspec b/node_modules/@bam.tech/react-native-image-resizer/react-native-image-resizer.podspec
index c7ba7d9..fb80b27 100644
--- a/node_modules/@bam.tech/react-native-image-resizer/react-native-image-resizer.podspec
+++ b/node_modules/@bam.tech/react-native-image-resizer/react-native-image-resizer.podspec
@@ -32,6 +32,6 @@ Pod::Spec.new do |s|
     s.dependency "RCTRequired"
     s.dependency "RCTTypeSafety"
     s.dependency "ReactCommon/turbomodule/core"
+    install_modules_dependencies(s)
   end
 end
-
