diff --git a/node_modules/@react-native-menu/menu/ios/MenuViewManager.mm b/node_modules/@react-native-menu/menu/ios/MenuViewManager.mm
index a974830..df40500 100644
--- a/node_modules/@react-native-menu/menu/ios/MenuViewManager.mm
+++ b/node_modules/@react-native-menu/menu/ios/MenuViewManager.mm
@@ -7,7 +7,7 @@
 #import "MenuView.h"
 #else
 // OLD ARCH
-#import <react_native_menu-Swift.h>
+#import <react_native_menu/react_native_menu-Swift.h>
 #endif
 
 @interface MenuViewManager : RCTViewManager
