diff --git a/node_modules/@tanstack/query-core/build/modern/infiniteQueryBehavior.cjs b/node_modules/@tanstack/query-core/build/modern/infiniteQueryBehavior.cjs
index 1891efa..54fe2f2 100644
--- a/node_modules/@tanstack/query-core/build/modern/infiniteQueryBehavior.cjs
+++ b/node_modules/@tanstack/query-core/build/modern/infiniteQueryBehavior.cjs
@@ -134,7 +134,7 @@ function getPreviousPageParam(options, { pages, pageParams }) {
   return pages.length > 0 ? options.getPreviousPageParam?.(pages[0], pages, pageParams[0], pageParams) : void 0;
 }
 function hasNextPage(options, data) {
-  if (!data) return false;
+  if (!data || (!Array.isArray(data) && !Object.keys(data).length)) return false;
   return getNextPageParam(options, data) != null;
 }
 function hasPreviousPage(options, data) {
diff --git a/node_modules/@tanstack/query-core/build/modern/infiniteQueryBehavior.js b/node_modules/@tanstack/query-core/build/modern/infiniteQueryBehavior.js
index 85e614f..85f8e4d 100644
--- a/node_modules/@tanstack/query-core/build/modern/infiniteQueryBehavior.js
+++ b/node_modules/@tanstack/query-core/build/modern/infiniteQueryBehavior.js
@@ -108,7 +108,7 @@ function getPreviousPageParam(options, { pages, pageParams }) {
   return pages.length > 0 ? options.getPreviousPageParam?.(pages[0], pages, pageParams[0], pageParams) : void 0;
 }
 function hasNextPage(options, data) {
-  if (!data) return false;
+  if (!data || (!Array.isArray(data) && !Object.keys(data).length)) return false;
   return getNextPageParam(options, data) != null;
 }
 function hasPreviousPage(options, data) {
