diff --git a/node_modules/react-native/Libraries/Network/FormData.js b/node_modules/react-native/Libraries/Network/FormData.js
index d1ec811..b06706f 100644
--- a/node_modules/react-native/Libraries/Network/FormData.js
+++ b/node_modules/react-native/Libraries/Network/FormData.js
@@ -55,13 +55,13 @@ class FormData {
     this._parts = [];
   }

-  append(key: string, value: FormDataValue) {
+  append(key: string, value: FormDataValue, filename?: string) {
     // The XMLHttpRequest spec doesn't specify if duplicate keys are allowed.
     // MDN says that any new values should be appended to existing values.
     // In any case, major browsers allow duplicate keys, so that's what we'll do
     // too. They'll simply get appended as additional form data parts in the
     // request body, leaving the server to deal with them.
-    this._parts.push([key, value]);
+    this._parts.push([key, value, filename]);
   }

   getAll(key: string): Array<FormDataValue> {
@@ -71,26 +71,26 @@ class FormData {
   }

   getParts(): Array<FormDataPart> {
-    return this._parts.map(([name, value]) => {
+    return this._parts.map(([name, value, filename]) => {
       const contentDisposition = 'form-data; name="' + name + '"';

-      const headers: Headers = {'content-disposition': contentDisposition};
+      const headers: Headers = { 'content-disposition': contentDisposition };

       // The body part is a "blob", which in React Native just means
       // an object with a `uri` attribute. Optionally, it can also
       // have a `name` and `type` attribute to specify filename and
       // content type (cf. web Blob interface.)
       if (typeof value === 'object' && !Array.isArray(value) && value) {
-        if (typeof value.name === 'string') {
-          headers['content-disposition'] += '; filename="' + value.name + '"';
-        }
-        if (typeof value.type === 'string') {
-          headers['content-type'] = value.type;
-        }
-        return {...value, headers, fieldName: name};
+        if (typeof filename === 'string' || typeof value.name === 'string')
+          headers['content-disposition'] +=
+            '; filename="' + (filename || value.name) + '"';
+
+        headers['content-type'] = value.type && typeof value.type === 'string' ? value.type : 'application/octet-stream';
+
+        return { ...value, headers, fieldName: name };
       }
       // Convert non-object values to strings as per FormData.append() spec
-      return {string: String(value), headers, fieldName: name};
+      return { string: String(value), headers, fieldName: name };
     });
   }
 }
