diff --git a/node_modules/react-native-parsed-text/src/ParsedText.d.ts b/node_modules/react-native-parsed-text/src/ParsedText.d.ts
index 5e63fbe..a1e0ff6 100644
--- a/node_modules/react-native-parsed-text/src/ParsedText.d.ts
+++ b/node_modules/react-native-parsed-text/src/ParsedText.d.ts
@@ -34,12 +34,12 @@ declare module 'react-native-parsed-text' {
     nonExhaustiveModeMaxMatchCount?: number;
   }
 
-  type ParseShape = DefaultParseShape | CustomParseShape;
+  export type ParseShape = DefaultParseShape | CustomParseShape;
 
   export interface ParsedTextProps extends TextProps {
     parse?: ParseShape[];
     childrenProps?: TextProps;
   }
 
-  export default class ParsedText extends Component<ParsedTextProps> {}
+  export default class ParsedText extends Component<ParsedTextProps> { }
 }
